### DISCUSSION TITLE
Gameplay preview on the song select screen

### DISCUSSION THREAD

**Discussion by DutchEllie** [ppy/osu#13136] [Ideas]
**Describe the new feature:** 
On the song select screen instead of the background picture of the map, there will be a gameplay preview of the map. This would be handy when scrolling through a large collection of maps, to choose the right one to play, and to be able to recognize the style of the map. This can of course be turned on or off in settings.

I am not sure if this would work, considering load times, but it would be an awesome feature! 

**Proposal designs of the feature:** 
Instead of the background picture of the map, there would be gameplay. Next to the setting to turn this on or off, you can dim the preview as well. 

(also, as seen below, idk how to add a label. I am terribly sorry)

**Comment by bdach** [comment]
This is probably quite a ways away; I'd say song select perf work and load time improvements are a must before this can even be considered. Due to those reasons marking as low priority for now.

**Comment by recognitionmark** [comment]
Maybe there will be options for the mapper to choose which part of the map to preview or a specific style of the map (take Notch Hell, HW can choose whether the map previews from the prev. point, the beginning, or one of the fast, long sliders, which are Notch Hells signature object.)

**Comment by peppy** [comment]
Removed last post as we don't really need / work with designs from other games. I'd rather not have content from other games posted on this repository.

If someone is looking to implement this we can provide a design to follow, but it wouldn't be based on other games.
